feat: Add universal support for encoding byte arrays

kristofbolyai opened the pull request:
EncodedByteBuffer is can be used for encoding and kind of byte[] data into formats like UTF-16 and base64. This is used as the base for the new gear chat encoding.

kristofbolyai committed:
feat: Add universal support for encoding byte arrays

EncodedByteBuffer is can be used for encoding and kind of byte[] data into formats like UTF-16 and base64. This is used as the base for the new gear chat encoding.

magicus commented on common/src/main/java/com/wynntils/core/encoding/EncodedByteBuffer.java:
```diff
@@ -4,25 +4,114 @@
+/**
+ * A buffer of bytes that can be encoded and decoded to various formats.
+ */
 public final class EncodedByteBuffer {
```
I must have missed when we got this class. It feels more "utils" than "core" to me.

kristofbolyai replied:
This PR targets a separate branch for gear encoding, so it doesn't exist on main.

magicus replied:
Ah. Nevertheless, maybe consider moving encoding stuff to utils? Or are you planning to include much more in this package, so it really will make sense to be a part of core?

kristofbolyai replied:
I've done some basic modeling in the PR this class was added, but I can agree with you. The main reason I put it in core is because I plan to use this class throughout Artemis for encoding different kind of data (like POIs maybe).

kristofbolyai replied:
I think it's best leaving it here until the original branch is ready and check back on this. If I add nothing else here, I am going to move it.

magicus replied:
Even fundamental types in Artemis (like Pair) have not resided in core. Maybe this was wrong, but if so we should move all fundamental types. Until we do that, I think this better belong with other more "general" types.

magicus replied:
Ok, sounds like a fair solution.

kristofbolyai replied:
I ended up moving the class here.

kristofbolyai committed:
chore: Move EncodedByteBuffer to utils package

This commit moves the `EncodedByteBuffer` class from the `core.encoding` package and places it into the `utils` package.
